pkg/eks/filecache.go

```diff
@@ -0,0 +1,227 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/gofrs/flock"
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// EksctlGlobalEnableCachingEnvName defines an environment property to enable the cache globally.
+	EksctlGlobalEnableCachingEnvName = "EKSCTL_ENABLE_CREDENTIAL_CACHE"
+	// EksctlCacheFilenameEnvName defines an environment property to configure where the cache file should live.
+	EksctlCacheFilenameEnvName = "EKSCTL_CREDENTIAL_CACHE_FILENAME"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+// Clock implements Now to return the current time.
+//counterfeiter:generate -o fakes/fake_clock.go . Clock
+type Clock interface {
+	Now() time.Time
+}
+
+// RealClock defines a clock using time.Now()
+type RealClock struct{}
+
+// Now returns the current time.
+func (r *RealClock) Now() time.Time {
+	return time.Now()
+}
```

Comment:
ahhh, a classic :smile:

Reply:
Whaaat, you don't like my Clock? :D :D :D

Reply:
I can replace it with a simple `func() time.Time` if you'd prefer that. :P

Reply:
no I love it! Its a classic